src/NetAnalyzers/UnitTests/Microsoft.NetCore.Analyzers/InteropServices/PlatformCompatibilityAnalyzer.ObsoletedInOSPlatformTests.cs

```diff
@@ -0,0 +1,457 @@
+                .WithArguments("Test.ObsoletedOnLinux4()", "'Linux' 4.1 and later", "'Linux'"));
+
+
+            var vbSource = @"
+Imports System
+Imports System.Runtime.Versioning
+Imports Mock
+Public Class Test
+    <SupportedOSPlatform(""Linux"")>
+    Public Sub M1()
+        ObsoletedOnWindows10()
+        {|#0:ObsoletedOnLinux()|} ' This call site is reachable on: 'Linux'. 'Public Sub ObsoletedOnLinux()' is obsoleted on: 'Linux' 4.1 and later.
+    End Sub
+
+    <ObsoletedInOSPlatform(""Windows10.1.1.1"")>
+    Public Sub ObsoletedOnWindows10()
+    End Sub
+    
+    <ObsoletedInOSPlatform(""Linux4.1"")>
+    Public Sub ObsoletedOnLinux()
+    End Sub
+End Class
+" + MockObsoletedAttributeVB;
+            await VerifyAnalyzerVBAsync(vbSource, VerifyVB.Diagnostic(PlatformCompatibilityAnalyzer.ObsoletedCsReachable).WithLocation(0)
+                .WithArguments("Public Sub ObsoletedOnLinux()", "'Linux' 4.1 and later", "'Linux'"));
+        }
+
+        [Fact]
+        public async Task ObsoletedWithMessageUrlCalledWarns()
+        {
+            var csSource = @"
+using System;
+using System.Runtime.Versioning;
+using Mock;
+
+public class Test
+{
+    public void CrossPlatform()
+    {
+        {|#0:ObsoletedWithMessageAndUrl()|}; // This call site is reachable on all platforms. 'Test.ObsoletedWithMessageAndUrl()' is obsoleted on: 'Windows' 10.1.1.1 and later, Use other method instead, http://www/look.for.more.info.
+        {|#1:ObsoletedWithMessage()|}; // This call site is reachable on all platforms. 'Test.ObsoletedWithMessage()' is obsoleted on: 'Windows' 10.1.1.1 and later, Use other method instead.
+        ObsoletedOnAndroid(); // Cross platform and android not in the MSBuild list so not warn
+    }
+    
+    [ObsoletedInOSPlatform(""android21.0"")]
+    public void ObsoletedOnAndroid() { }
+
+    [ObsoletedInOSPlatform(""Windows10.1.1.1"", ""Use other method instead"", Url = ""http://www/look.for.more.info"")]
+    public void ObsoletedWithMessageAndUrl() { }
+    [ObsoletedInOSPlatform(""Windows10.1.1.1"", ""Use other method instead"")]
+    public void ObsoletedWithMessage() { }
+}" + MockObsoletedAttributeCS;
+            await VerifyAnalyzerCSAsync(csSource, s_msBuildPlatforms, VerifyCS.Diagnostic(PlatformCompatibilityAnalyzer.ObsoletedCsAllPlatforms).WithLocation(0).
+                    WithArguments("Test.ObsoletedWithMessageAndUrl()", "'Windows' 10.1.1.1 and later, Use other method instead http://www/look.for.more.info"),
+                  VerifyCS.Diagnostic(PlatformCompatibilityAnalyzer.ObsoletedCsAllPlatforms).WithLocation(1).
+                    WithArguments("Test.ObsoletedWithMessage()", "'Windows' 10.1.1.1 and later, Use other method instead"));
+        }
+
+        [Fact]
+        public async Task ObsoletedAPIsCAlledFromDifferentCallsite()
+        {
+            var csSource = @"
+using System;
+using System.Runtime.Versioning;
+using Mock;
```

Comment:
Why are some of these using the `Mock` attributes?

Reply:
For now the new `ObsoletedInOSPlatform` attribute is not available for the repo (and the update that added message to the `UnsupportedOSPlatform` attribute), not sure when it will become available